packages/clerk-js/src/ui/common/withRedirectToHome.tsx

```diff
@@ -1,35 +1,67 @@
 import { SignInProps, SignUpProps } from '@clerk/types';
-import React, { useContext } from 'react';
+import React from 'react';
 
-import { useEnvironment } from '../contexts';
-import { CoreSessionContext } from '../contexts/CoreSessionContext';
+import { canShowOrganizationProfile, canShowSignIn, canShowSignUp, canShowUserProfile } from '../../utils';
+import { useCoreClerk, useEnvironment } from '../contexts';
 import { useNavigate } from '../hooks';
 
 export function withRedirectToHome<P extends SignInProps | SignUpProps | { continueExisting?: boolean }>(
   Component: React.ComponentType<P>,
-  displayName?: string,
+  when: 'signIn' | 'signUp' | 'userProfile' | 'organizationProfile',
```

Comment:
If we still need this type after the refactorings discussed below, I'd look at the `appearance` or `Component.tsx` types because I'm almost sure we have this already

Reply:
The type introduced is that of the requirement that needs to be met, so I'm resolving this.